programs/conditional_vault/src/instructions/initialize_conditional_vault.rs

```diff
@@ -53,26 +31,67 @@ pub struct InitializeConditionalVault<'info> {
     pub system_program: Program<'info, System>,
 }
 
-impl InitializeConditionalVault<'_> {
-    pub fn handle(ctx: Context<Self>, args: InitializeConditionalVaultArgs) -> Result<()> {
+impl<'info, 'c: 'info> InitializeConditionalVault<'info> {
+    pub fn handle(ctx: Context<'_, '_, 'c, 'info, Self>) -> Result<()> {
         let vault = &mut ctx.accounts.vault;
 
-        let InitializeConditionalVaultArgs {
-            settlement_authority,
-        } = args;
+        let decimals = ctx.accounts.underlying_token_mint.decimals;
+
+        let remaining_accs = &mut ctx.remaining_accounts.iter();
+
+        let expected_num_conditional_tokens = ctx.accounts.question.num_outcomes();
+        let mut conditional_token_mints = vec![];
+
+        let mint_lamports = Rent::get()?.minimum_balance(Mint::LEN);
+        for i in 0..expected_num_conditional_tokens {
+            let (conditional_token_mint_address, pda_bump) = Pubkey::find_program_address(
+                &[b"conditional_token", vault.key().as_ref(), &[i as u8]],
+                ctx.program_id,
+            );
+
+            let conditional_token_mint = next_account_info(remaining_accs)?;
+            require_eq!(conditional_token_mint.key(), conditional_token_mint_address);
+
+            conditional_token_mints.push(conditional_token_mint_address);
+
+            let cpi_program = ctx.accounts.system_program.to_account_info();
+            let cpi_accounts = system_program::CreateAccount {
+                from: ctx.accounts.payer.to_account_info(),
+                to: conditional_token_mint.to_account_info(),
+            };
+            let vault_key = vault.key();
+            let seeds = &[
+                b"conditional_token",
+                vault_key.as_ref(),
+                &[i as u8],
+                &[pda_bump],
+            ];
+            let signer = &[&seeds[..]];
+            let cpi_ctx = CpiContext::new_with_signer(cpi_program, cpi_accounts, signer);
+
+            system_program::create_account(
```

Comment:
https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/programs/system/src/system_processor.rs#L161
when accounts already have lamports, create_account will fail
attackers may send a lamport to your conditional accounts and make initialization impossible.

Reply:
that's retarded, okay I guess I will do what anchor does: https://github.com/coral-xyz/anchor/blob/340e9c13f42191a1caa6092d811ed930a64c1f7b/lang/syn/src/codegen/accounts/constraints.rs#L1603-L1627

Reply:
fixed: 166de60ab202d30c2f1274b0abbd8cd8dbdf5df8